runner/app/routes/transformers.py

```diff
@@ -0,0 +1,160 @@
+        str,
+        Field(
+            description=(
+                "The transformer model to use for the task. E.g. 'openai/whisper-base'."
+            ),
+        ),
+    ]
+    input: Annotated[
+        Union[str, HttpUrl],
+        Field(
+            description=(
+                "The input data to be transformed. Can be string or an url to a file."
+            ),
+        ),
+    ]
+    pipeline_params: Dict[str, Any] = Field(
+        default_factory=dict,
+        description="Additional keyword arguments to pass to the transformer pipeline during inference. E.g. {'return_timestamps': True, 'max_length': 50}.",
+    )
+
+
+class InferenceResponse(BaseModel):
+    """Response model for transformer inference."""
+
+    output: Any = Field(
+        ..., description="The output data transformed by the transformer pipeline."
+    )
+
+
+RESPONSES = {
+    status.HTTP_200_OK: {
+        "content": {
+            "application/json": {
+                "schema": {
+                    "x-speakeasy-name-override": "data",
+                }
+            }
+        },
+    },
+    status.HTTP_400_BAD_REQUEST: {"model": HTTPError},
+    status.HTTP_401_UNAUTHORIZED: {"model": HTTPError},
+    status.HTTP_500_INTERNAL_SERVER_ERROR: {"model": HTTPError},
+}
+
+
+@router.post(
+    "/transformers",
+    response_model=InferenceResponse,
+    responses=RESPONSES,
+    description="Perform inference using a Hugging Face transformer model.",
+    operation_id="genTransformers",
+    summary="Transformers",
+    tags=["generate"],
+    openapi_extra={"x-speakeasy-name-override": "transformers"},
+)
+@router.post("/transformers/", responses=RESPONSES, include_in_schema=False)
+async def transformers(
+    request: InferenceRequest,
+    token: HTTPAuthorizationCredentials = Depends(HTTPBearer(auto_error=False)),
+):
+    auth_token = os.environ.get("AUTH_TOKEN")
+    if auth_token:
+        if not token or token.credentials != auth_token:
+            return JSONResponse(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                headers={"WWW-Authenticate": "Bearer"},
+                content=http_error("Invalid bearer token."),
+            )
+
+    if not request.task and not request.model_name:
+        raise JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content=http_error("Either 'task' or 'model_name' must be provided."),
+        )
+    if not request.input:
+        raise JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content=http_error("'input' field is required."),
+        )
+
+    torch_device = get_torch_device()
+
+    # Initialize the pipeline with the specified task and model ID.
+    pipeline_kwargs = {}
+    if request.task:
+        pipeline_kwargs["task"] = request.task
+    if request.model_name:
+        pipeline_kwargs["model"] = request.model_name
+    try:
+        pipe = pipeline(device=torch_device, **pipeline_kwargs)
```

Comment:
I'm haven't checked whether models are removed from the GPU afterward.

Reply:
@rickstaa I think either we can perform cache cleaning with `empty_cache()` or use pipeline in context manager: `with pipeline(...)`